### ISSUE TITLE
Snapmod folder contents get deleted after a while 

### ISSUE THREAD

**Issue by jkooper32** [rodit/SnapMod#160]
Hello there, 

I save pictures and videos sometimes to the default location, which is the Snapmod folder but after a few days the whole folder is gone and all media in it gets deleted. Is this a common issue with the mod or maybe there's something wrong with my configuration? 

Thanks

**Comment by RevealedSoulEven** [comment]
Maybe it gets deleted when you try to uninstall SnapMod or if it's getting deleted automatically, make sure you are not cleaning your phone using any phone cleaner apps or space-saving apps. Also make sure to make a copy in this case, or change the default SnapMod saving directory.

**Comment by rodit** [comment]
Yes, unfortunately this is a known issue with Android's download manager. It deletes downloaded files that are not in `Internal Storage/Downloads` after a while. I only became aware of this very recently. You can either change the download path to be `$Downloads/SnapMod/...` or rename/move the files after they are downloaded to solve this. Alternatively, you can disable the 'Use Android Download Manager' option, although this may not work on Android 11/12+. I will also look into ways of preventing them from being deleted.

I will leave this issue open until this is safely resolved.

**Comment by Magna-Graecia** [comment]
How long is "a while"? I have a snap saved in the SnapMod folder since May 5th, and it still hasn't been deleted. I also have Android download manager enabled.

**Comment by rodit** [comment]
I assume it depends on device/android implementation, maybe even on mhow much free space you have. Just to be safe, I'd recommend renaming/moving the files every now and again.

**Comment by tycoon420** [comment]
exactly same thing happened to me 2 times already(my phone internal is 512Gb with 350Gb free space), so space wasn't the issue. The issue maybe due to old magisk. After i switched to zygisk i didn't encounter the deleting problem. i have switched to zygisk just 4 day ago. let's see if the pics n videos get deleted now.

**Comment by rodit** [comment]
The issue is described [here](https://medium.com/@mkaflowski/disappearing-files-downloaded-by-downloadmanager-7c9ee5c6a66a), it typically will happen after 7 days.

**Comment by LiqwidSw** [comment]
>The issue maybe due to old magisk. After i switched to zygisk i didn't encounter the deleting problem. i have switched to zygisk just 4 day ago. let's see if the pics n videos get deleted now.

Zygisk is not safe from the issue. Been using SnapMod for about month, just noticed this week I only had snaps from the last 7 days saved. I thought I had somehow deleted the snaps by mistake until I checked this thread.
